src/store/channels-list/saga.ts

```diff
@@ -366,6 +366,22 @@ export function* userLeftChannel(channelId, userId) {
   }
 }
 
+// TODO: we can remove this function and simply use the "lastMessage.createdAt" property
+// look into https://github.com/zer0-os/zOS/pull/1063
+const conversationWithLatestMessage = (conversations) =>
+  conversations.reduce((latest, current) => {
+    if (current.messages && current.messages.length > 0) {
+      const latestMessage = current.messages.reduce((prevMsg, currentMsg) => {
+        return currentMsg.createdAt > prevMsg.createdAt ? currentMsg : prevMsg;
+      });
+      if (!latest || latestMessage.createdAt > latest.messages[0].createdAt) {
+        current.latestMessage = latestMessage;
```

Comment:
Do we have to set this attribute at all? We're returning the conversation.

Reply:
updated but i still needed this for sorting. Created a `deepCopy` and did it on that.